huaweicloud/services/live/resource_huaweicloud_live_transcoding.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	response, err := client.ShowTranscodingsTemplate(&model.ShowTranscodingsTemplateRequest{
+		Domain:  domain,
+		AppName: &appName,
+	})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live transcoding")
+	}
+
+	if response.Templates == nil || len(*response.Templates) != 1 {
+		return diag.Errorf("error retrieving Live transcoding")
+	}
+	r := *response.Templates
+	detail := r[0]
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("domain_name", response.Domain),
+		d.Set("app_name", detail.AppName),
+		setTemplatesToState(d, detail.QualityInfo),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceTranscodingUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	transcodingParams, err := buildTranscodingParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = client.UpdateTranscodingsTemplate(&model.UpdateTranscodingsTemplateRequest{
+		Body: transcodingParams,
+	})
+
+	if err != nil {
+		return diag.Errorf("error updating Live transcoding: %s", err)
+	}
+
+	return resourceTranscodingRead(ctx, d, meta)
+}
+
+func resourceTranscodingDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	deleteOpts := &model.DeleteTranscodingsTemplateRequest{
+		Domain:  d.Get("domain_name").(string),
+		AppName: d.Get("app_name").(string),
+	}
+	_, err = client.DeleteTranscodingsTemplate(deleteOpts)
+	if err != nil {
+		return diag.Errorf("error deleting Live transcoding: %s", err)
+	}
+
+	return nil
+}
+
+func buildTranscodingParams(d *schema.ResourceData) (*model.StreamTranscodingTemplate, error) {
+	var codec model.QualityInfoCodec
+	if err := codec.UnmarshalJSON([]byte(d.Get("video_encoding").(string))); err != nil {
+		return nil, fmt.Errorf("error parsing the argument %q: %s", "video_encoding", err)
+	}
+
+	var hdlb model.QualityInfoHdlb
+	if v, ok := d.GetOk("low_bitrate_hd"); ok {
+		if err := hdlb.UnmarshalJSON([]byte(v.(string))); err != nil {
+			return nil, fmt.Errorf("error parsing the argument %q: %s", "low_bitrate_hd", err)
+		}
+	}
+
+	templates := d.Get("templates").([]interface{})
+	qualityInfo := make([]model.QualityInfo, len(templates))
+	for i, v := range templates {
+		template := v.(map[string]interface{})
+
+		width := template["width"].(int)
+		height := template["height"].(int)
+
+		errFmt := "expected %s to be in the range (%d - %d) and must be a multiple of %d when " +
+			"video_encoding is %s, got %d"
+
+		if codec == model.GetQualityInfoCodecEnum().H264 {
+			if width < 32 || width > 3840 || width%2 != 0 {
```

Comment:
Nits: is it possible to add this as a validation func?  then we can get error on `terrafrom plan` stage.

Reply:
ValidateFunc just has two arguments:  key and value. we cannot get other attribute value